test/integration/metrics_test.go

```diff
@@ -280,6 +284,90 @@ func TestAllMetricsPresent(t *testing.T) {
 	})
 }
 
+// As mentioned in the docker API docs (https://docs.docker.com/engine/api/v1.44/#tag/Container/operation/ContainerStats)
+// the only field set in `blkio_stats` is `io_service_bytes_recursive`.
+func TestBlkIOMetrics(t *testing.T) {
+	cases := []struct {
+		Name          string
+		MockStats     types.StatsJSON
+		ExpectedBlkIO biz.BlkIO
+	}{
+		{
+			Name: "With IoServiceBytesRecursive and IoServicedRecursive",
+			MockStats: types.StatsJSON{
+				Stats: types.Stats{
+					BlkioStats: types.BlkioStats{
+						IoServiceBytesRecursive: []types.BlkioStatEntry{
+							{Op: "Read", Value: 5885952},
+							{Op: "Write", Value: 45056},
+						},
+						IoServicedRecursive: []types.BlkioStatEntry{
+							{Op: "Read", Value: 39},
+							{Op: "Write", Value: 11},
+						},
+					},
+				},
+			},
+			ExpectedBlkIO: biz.BlkIO{
+				TotalReadCount:  pointerToFloat64(39),
+				TotalWriteCount: pointerToFloat64(11),
+				TotalReadBytes:  pointerToFloat64(5885952),
+				TotalWriteBytes: pointerToFloat64(45056),
+			},
+		},
+		{
+			Name: "Without IoServiceBytesRecursive and IoServicedRecursive",
+			MockStats: types.StatsJSON{
+				Stats: types.Stats{
+					BlkioStats: types.BlkioStats{},
+				},
+			},
+			ExpectedBlkIO: biz.BlkIO{
+				TotalReadCount:  nil,
+				TotalWriteCount: nil,
+				TotalReadBytes:  nil,
+				TotalWriteBytes: nil,
+			},
+		},
+		{
+			Name: "With Only IoServiceBytesRecursive",
+			MockStats: types.StatsJSON{
+				Stats: types.Stats{
+					BlkioStats: types.BlkioStats{
+						IoServiceBytesRecursive: []types.BlkioStatEntry{
+							{Op: "Read", Value: 5885952},
+							{Op: "Write", Value: 45056},
+						},
+					},
+				},
+			},
+			ExpectedBlkIO: biz.BlkIO{
+				TotalReadCount:  nil,
+				TotalWriteCount: nil,
+				TotalReadBytes:  pointerToFloat64(5885952),
+				TotalWriteBytes: pointerToFloat64(45056),
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.Name, func(t *testing.T) {
+			client := mockDockerStatsClient{}
+			client.On("ContainerStats", mock.Anything).Return(tc.MockStats)
+			dockerAPIFetcher := dockerapi.NewFetcher(&client)
+
+			storer := inMemoryStorerWithPreviousCPUState()
+			inspector := NewInspectorMock(InspectorContainerID, InspectorPID, 2)
+
+			metrics := biz.NewProcessor(storer, dockerAPIFetcher, inspector, 0)
+
+			sample, err := metrics.Process(InspectorContainerID)
```

Comment:
couldn't the test use the SampleAll to cover more code and check the metrics in the sample? perhaps is not so hard at this point

Reply:
As discussed, this would require a refactor since `ContainerSampler` is living in the `nri` package and has all the fields unexported. So, we cannot directly use it in our integration tests. For more context, it's used as follows:
```golang
sampler := ContainerSampler{
    metrics: biz.NewProcessor(mStore, nil, mocker, 30*time.Minute),
    docker:  mocker,
    store:   mStore,
}
err = sampler.SampleAll(context.Background(), i, info)
```